models/types/schedule_type.rb

```diff
@@ -18,24 +18,25 @@
 
 class ScheduleType
   def self.type_cast(value)
-    return_value = if !value.nil?
-                    if /\A([0-9]+):([0-9]{2}):([0-9]{2})\z/ =~ value.to_s
-                      3600 * Regexp.last_match(1).to_i + 60 * Regexp.last_match(2).to_i + Regexp.last_match(3).to_i
-                    elsif /\A([0-9]+):([0-9]{2})\z/ =~ value.to_s
-                      3600 * Regexp.last_match(1).to_i + 60 * Regexp.last_match(2).to_i
-                    elsif /\A[0-9]+\.{0,1}[0-9]*\z/ =~ value.to_s
-                      value.to_i
-                    elsif value.is_a?(Integer) || value.is_a?(Float)
-                      value.to_i
-                    else
-                      log 'error', level: :error
-                      raise ArgumentError, 'Invalid Time value'
-                    end
-                  end
+    return_value =
+      if value
+        if /\A([0-9]+):([0-9]{2}):([0-9]{2})\z/ =~ value.to_s
+          3600 * Regexp.last_match(1).to_i + 60 * Regexp.last_match(2).to_i + Regexp.last_match(3).to_i
+        elsif /\A([0-9]+):([0-9]{2})\z/ =~ value.to_s
+          3600 * Regexp.last_match(1).to_i + 60 * Regexp.last_match(2).to_i
+        elsif /\A[0-9]+\.{0,1}[0-9]*\z/.match?(value.to_s)
+          value.to_i
+        elsif value.is_a?(Integer) || value.is_a?(Float)
+          value.to_i
+        else
+          log 'error', level: :error
+          raise ArgumentError.new('Invalid Time value')
+        end
+      end
 
     if return_value&.negative?
       log 'error', level: :error
-      raise ArgumentError, 'Invalid Time value'
+      raise ArgumentError.new('Invalid Time value')
```

Comment:
We can be more precise here since we know the reason
```suggestion
      raise ArgumentError.new('Time value cannot be negative')
```